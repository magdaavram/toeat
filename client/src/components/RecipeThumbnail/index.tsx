import React from 'react';
import styled from 'styled-components';
import DifficultyLevel from 'components/DifficultyLevel';
import CookingTime from 'components/CookingTime';
import { Link } from 'react-router-dom';
import Image from 'components/Image';
import { IRecipe } from '../../api/Recipe';

const Container = styled.div`
  max-width: 400px;
`;

const StyledImage = styled(Image)`
  width: 100%;
  height: 250px;
`;

const DetailsContainer = styled.div`
  display: flex;
  align-items: center;
  margin: 9px 0 0 3px;
  color: var(--color--medium-purple);
  font-size: var(--font-size--small);
  font-family: 'Roboto Medium';
`;

const Detail = styled.div`
  display: flex;
  align-items: center;
  margin-right: 27px;
`;

const Title = styled.span`
  display: inline-block;
  width: 100%;
  padding: 0;
  margin: 6px 0 9px 0;
  color: var(--color--dark-purple);
  font-size: var(--font-size--large);
  font-family: 'Suisse SemiBold';
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover {
    color: var(--color--burgundy);
  }
`;

const RecipeThumbnail = (recipe: IRecipe) => {
  const { id, imageUrl, title, duration, difficultyLevel, course } = recipe;

  return (
    <Container>
      <Link to={'/recipe/' + id} title={title}>
        <StyledImage url={imageUrl} course={course} className="thumbnail-image" />
      </Link>
      <DetailsContainer>
        <Detail>
          <CookingTime duration={duration} hasIcon={true} />
        </Detail>
        <Detail>
          <DifficultyLevel level={difficultyLevel} hasIcon={true} />
        </Detail>
      </DetailsContainer>
      <Link to={'/recipe/' + id} title={title}>
        <Title>{title}</Title>
      </Link>
    </Container>
  );
};

export default RecipeThumbnail;
