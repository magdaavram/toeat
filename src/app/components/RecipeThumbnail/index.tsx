import React from 'react';
import styled from 'styled-components';
import DifficultyLevel from '../DifficultyLevel';
import CookingTime from '../CookingTime';

interface IImageProps {
  url: string;
}

interface IProps {
  imageUrl: string;
  title: string;
  duration: number;
  difficultyLevel: number;
}

const Container = styled.div`
  max-width: 400px;
`;

const Image = styled.div((props: IImageProps) => ({
  width: '100%',
  height: '250px',
  background: `url(${props.url}) no-repeat center`,
  backgroundSize: 'cover',
}));

const ImageLink = styled.a`
  display: inline-block;
  width: 100%;
  height: 100%;
`;

const DetailsContainer = styled.div`
  display: flex;
  align-items: center;
  margin-top: 6px;
  color: var(--color--medium-purple);
  font-size: var(--font-size--small);
  font-family: 'Roboto Medium';
`;

const Detail = styled.div`
  display: flex;
  align-items: center;
  margin-right: 27px;
`;

const Title = styled.a`
  display: inline-block;
  width: 100%;
  padding: 0;
  margin: 6px 0 9px 0;
  text-decoration: none;
  color: var(--color--dark-purple);
  font-size: var(--font-size--large);
  font-family: 'Suisse SemiBold';
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover {
    color: var(--color--burgundy);
  }
`;

const RecipeThumbnail = (props: IProps) => {
  const { imageUrl, title, duration, difficultyLevel } = props;

  return (
    <Container>
      <Image url={imageUrl}>
        <ImageLink href={'/#'} title={title} />
      </Image>
      <DetailsContainer>
        <Detail>
          <CookingTime duration={duration} hasIcon={true} />
        </Detail>
        <Detail>
          <DifficultyLevel level={difficultyLevel} hasIcon={true} />
        </Detail>
      </DetailsContainer>
      <Title href={'/#'} title={title}>
        {title}
      </Title>
    </Container>
  );
};

export default RecipeThumbnail;
