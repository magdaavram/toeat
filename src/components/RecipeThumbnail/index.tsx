import React from 'react';
import styled from 'styled-components';
import DifficultyLevel from 'components/DifficultyLevel';
import CookingTime from 'components/CookingTime';
import { Link } from 'react-router-dom';

interface IImageProps {
  url: string;
}

interface IProps {
  imageUrl: string;
  title: string;
  duration: number;
  difficultyLevel: number;
}

const Container = styled.div`
  max-width: 400px;
`;

const ImageContainer = styled.div((props: IImageProps) => ({
  width: '100%',
  height: '250px',
  background: `url(${props.url}) no-repeat center`,
  backgroundSize: 'cover',
}));

const DetailsContainer = styled.div`
  display: flex;
  align-items: center;
  margin: 9px 0 0 3px;
  color: var(--color--medium-purple);
  font-size: var(--font-size--small);
  font-family: 'Roboto Medium';
`;

const Detail = styled.div`
  display: flex;
  align-items: center;
  margin-right: 27px;
`;

const Title = styled.span`
  display: inline-block;
  width: 100%;
  padding: 0;
  margin: 6px 0 9px 0;
  color: var(--color--dark-purple);
  font-size: var(--font-size--large);
  font-family: 'Suisse SemiBold';
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover {
    color: var(--color--burgundy);
  }
`;

const RecipeThumbnail = (props: IProps) => {
  const { imageUrl, title, duration, difficultyLevel } = props;

  return (
    <Container>
      <Link to="/recipe" title={title}>
        <ImageContainer url={imageUrl} />
      </Link>
      <DetailsContainer>
        <Detail>
          <CookingTime duration={duration} hasIcon={true} />
        </Detail>
        <Detail>
          <DifficultyLevel level={difficultyLevel} hasIcon={true} />
        </Detail>
      </DetailsContainer>
      <Link to="/recipe" title={title}>
        <Title>{title}</Title>
      </Link>
    </Container>
  );
};

export default RecipeThumbnail;
