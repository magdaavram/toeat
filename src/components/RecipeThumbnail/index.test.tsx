import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { Router, MemoryRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Routes from 'routes';
import RecipeThumbnail from './index';

it('should render correctly', () => {
  const tree = create(
    <MemoryRouter>
      <RecipeThumbnail imageURL={'image.png'} title={'Pasta'} duration={78} difficultyLevel={2} />
    </MemoryRouter>
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should change URL when clicking on image', () => {
  const history = createMemoryHistory();
  const wrapper = mount(
    <Router history={history}>
      <RecipeThumbnail imageURL={'image.png'} title={'Pasta'} duration={78} difficultyLevel={2} />
      <Routes />
    </Router>
  );

  wrapper.find('a').at(0).simulate('click', { button: 0 });
  expect(history.location.pathname).toEqual('/recipe');
});

it('should change URL when clicking on title', () => {
  const history = createMemoryHistory();
  const wrapper = mount(
    <Router history={history}>
      <RecipeThumbnail imageURL={'image.png'} title={'Pasta'} duration={78} difficultyLevel={2} />
      <Routes />
    </Router>
  );

  wrapper.find('a').at(1).simulate('click', { button: 0 });
  expect(history.location.pathname).toEqual('/recipe');
});
