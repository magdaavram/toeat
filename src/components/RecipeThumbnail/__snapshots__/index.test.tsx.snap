// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="sc-fzoLag hhVJqd"
>
  <a
    href="/recipe"
    onClick={[Function]}
    title="Pasta"
  >
    <div
      className="sc-fzoXzr kTgEsu"
    />
  </a>
  <div
    className="sc-fzpjYC eNSNGf"
  >
    <div
      className="sc-fznxsB jSNgKC"
    >
      <div
        className="sc-fzoyAV YXuwE"
      >
        <svg
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <image
              height="20"
              href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAMAAAAM7l6QAAAAAXNSR0IB2cksfwAAAOpQTFRFAAAASk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pfrZ05wAAAE50Uk5TAAESOGim1O73I3G94/z/CFnJ/dKzpySR+sdwE5Cs9dx2IAar4m4JVQREiz7C9g1ByvmuEAJSGJnr+A+Wz2Kk8B0Hc8MLh+BbJm93yCJYgSwx+QAAAWlJREFUeJx9k+l2gjAQhQMIBJBBgigCoiCiXe1i99rW7nV5/9dpWQMe6fxKzpdzM8sdhNJgWK7BCyLGosA3OJZB1ZBkpalCEmpTkaUK1Fo6BiBG2zTbBgHAekujtNO1gPRsJ705do+A5RYCnb4Hg6Ef5PfAHw7AG2Vc63oQjqPyZ9E4BM9N9VsWhJNpNdPpJASrFVck6TAYZzQ6ODxKa5qOB6BLDGJlTIa5Mnd8wp9mL4cEyyziFOj5uebs7PziMjv7PVA41GiCHezDgQxhA/EqcdA+jByi8kgAA+3HyAABidCuw20QEQazDpuAK3h+Zezisvj1ze1d/74iXkpt/uDB44K2N06NFuY8ATy/0ESSwoq2LF8t8ragNG1L0dT3D/j8oh9nTS1Gwnz/zJaURiuCbbY6UBrxQNebeLi1dtgmR83910xIGu1acVWy4h93K0aWq0beswbrbWkNULJEYb5EoWJvdgqpWcFfq5ArMTOrWiwAAAAASUVORK5CYII="
              id="clock-icon"
              width="20"
            />
          </defs>
          <use
            href="#clock-icon"
            x="0"
            y="0"
          />
        </svg>
      </div>
      1h 18 min
    </div>
    <div
      className="sc-fznxsB jSNgKC"
    >
      <div
        className="sc-fzqNJr skGAO"
      >
        <svg
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <image
              height="20"
              href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAcCAMAAABBJv+bAAAAAXNSR0IB2cksfwAAAexQTFRFAAAASk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk1oSU1qSk5pSk5pSk5pR1JmUFBwSk5pSk5pSU1pSU9qSk5pSk5pRlFoTk5iSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pS0tpSExrSk1oSk5qS01qSU1oTExmSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSU9qSU5pSU1pS05qSk9pSk5pSk5pSk5pSk1pTFBoAAAASk5pSk5pSk5pTU1mSk1pSUltSk5pSk5pTExrSVBmSk5pSk5pSk5pSk5pSk5pSk1qSk5nSk5pSk5pSk5pSk5pSk5pSk5pSk5qSU9qQFVqR1VjSk9pSU5oSk5pS09oRkZ0S0tpS01p0U8fqgAAAKR0Uk5TAFhSFODcEXK9xWoN5UBH4gogQmWMv7gDBL68i2RBH8Puro9xJSiQr+/hYGY9Ot9s3jU/REzGNDvEBTLMFRorVdFiws5h00/0Ww8QXsgOZ11GXUYZ6dkZEHr/el4W2hYNIkNojrqKYx4iQ466Y0Iez/b+9bZN7ElN7ElB4O02Ae02Ad3JBxTJB+QjMiP8+7vHRWBFgPdXDBL4gFcMEmRiC0cLEWbeEiVEAAABdklEQVR4nGNgQAGMTAx4ADMLKxseaXYOTi7cstw8vHz8AlilBIWERUTFxCUkpaRlZOUwpOUVFJWUVRgYVNnV1DU4MKSFNLXEISxxbRYdTNN19fTB8uLsBobYbDdiNQZRJqy6WB0nYGoGoswtLLFKW1nbSMhoidta22GVtndwdBB1cnZxdcMqze5u6uHp5e3jro9VWlbGF0R5+mF6yz8AlR8YhMwLDgn1QuaHhYSGI3EjIiO9kaWjIiOjEbwYhcjIWG4EPw7EjwczExKTki0igSAlNU0onYEhIzMpC8zPTs3JzWPIL4iEgcKiYgZ0fkkpnBvJX8bAUF6BxK8EehcuzyILsq0KLl9dBeKX1UB4mrV1YNdU1kP5DY0gLnMThNvcAnF3K5Tf1g7mWkF1d3RCpLuguju6wVytnsjevv6eyIIJEOmJMP4kMHfyFM6p06bL5M9whEjPnMI5a/acufkz5oG5fN7zQRSbdi1EekH0QhC1aHEDAwMAujBwrZPrJrUAAAAASUVORK5CYII="
              id="medium-icon"
              width="20"
            />
          </defs>
          <use
            href="#medium-icon"
            x="0"
            y="1"
          />
        </svg>
      </div>
      Medium
    </div>
  </div>
  <a
    href="/recipe"
    onClick={[Function]}
    title="Pasta"
  >
    <span
      className="sc-fznJRM dNLTMg"
    >
      Pasta
    </span>
  </a>
</div>
`;
